.github/workflows/wheels.yml

```diff
@@ -75,9 +79,9 @@ jobs:
           CIBW_TEST_SKIP: "*-macosx_arm64"
           MACOSX_DEPLOYMENT_TARGET: ${{ matrix.macosx_deployment_target }}
 
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4
         with:
-          name: dist
+          name: ${{ matrix.artifact_name }}
```

Comment:
Can we re-use existing matrix variables instead of introducing a new `artifact_name`?

The tricky bit might be differentiating between these two:

```yaml
          - name: "manylinux2014 and musllinux x86_64"
            os: ubuntu-latest
            archs: x86_64
            artifact_name: "manylinux2014_musllinux"
          - name: "manylinux_2_28 x86_64"
            os: ubuntu-latest
            archs: x86_64
            build: "*manylinux*"
            manylinux: "manylinux_2_28"
            artifact_name: "manylinux_2_28"
```

If we do `wheel-${{ matrix.os }}-${{ matrix.archs }}-${{ matrix.manylinux }}`, does that error or do we get something like this?

* `wheel-ubuntu-latest-x86_64`
* `wheel-ubuntu-latest-x86_64-manylinux_2_28`

Reply:
I've pushed a commit to use `${{ matrix.os }}-${{ matrix.archs }}-${{ matrix.manylinux }}`.

We get
- macos-latest-x86_64-
- macos-latest-arm64-
- ubuntu-latest-x86_64-
- ubuntu-latest-x86_64-manylinux_2_28

Reply:
Thanks!

Since my last suggestion, and based on lots of feedback about the breaking `@v4` changes, GitHub have just added `pattern:` and `merge-multiple:` inputs:

https://github.com/actions/upload-artifact/issues/472#issuecomment-1861571655

Shall we try that? We'll need a common prefix/pattern for the sdist and all the wheel artifacts.

Reply:
Thanks for finding that. I've updated the PR.